### PR TITLE
GetASNInt(): check for invalid padding on negative integer.

### PR DESCRIPTION
**Pull request by douzzer**
re ZD ticket 10840

### CODE REVIEW

**Comment by douzzer** [comment]
I've just updated the ticket with a pointer to the PR and a request for feedback.

File: wolfcrypt/src/asn.c [hunk 1]
```diff
@@ -464,6 +464,17 @@ static int GetASNInt(const byte* input, word32* inOutIdx, int* len,
         return ret;
 
     if (*len > 0) {
+
+#ifndef WOLFSSL_ASN_INT_LEAD_0_ANY
```
Comment (SparkiDev) [inline wolfcrypt/src/asn.c:468]: Why does this here?
This function is a generic ASN.1 integer parsing function.
Negative numbers are allowed, just not in X.509 serial numbers.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/asn.c (hunk 1 above)
Comment (douzzer) [reply wolfcrypt/src/asn.c:468]: It's only adding X.690-pursuant rejection of frivolously padded negative numbers, just as frivolously padded positive numbers were already rejected.  In both cases (as before in the case of positive), defining `WOLFSSL_ASN_INT_LEAD_0_ANY` inhibits the rejection.

More to the point: negative numbers are still allowed by the routine as modified, as they should be.

**Review by SparkiDev — APPROVED** [review]

**Comment by douzzer** [comment]
@toddouska Guido gave it a workout and [is happy with it](https://wolfssl.zendesk.com/agent/tickets/10840).

**Pull request merged**